server/service/client_live_query.go

```diff
@@ -131,10 +140,25 @@ func (c *Client) LiveQuery(query string, labels []string, hosts []string) (*Live
 				Type string          `json:"type"`
 				Data json.RawMessage `json:"data"`
 			}{}
-			err := conn.ReadJSON(&msg)
-			if err != nil {
-				resHandler.errors <- errors.Wrap(err, "receive ws message")
+
+			doneReadingChan := make(chan error)
+
+			go func() {
+				doneReadingChan <- conn.ReadJSON(&msg)
+			}()
+
+			select {
+			case <-ctx.Done():
+				return
+			case err := <-doneReadingChan:
+				if err != nil {
+					resHandler.errors <- errors.Wrap(err, "receive ws message")
+					if errors.Is(err, websocket.ErrCloseSent) {
+						return
+					}
+				}
 			}
+			close(doneReadingChan)
```

Comment:
Does this simplification work?

```suggestion
			select {
			case <-ctx.Done():
				return
			case err := <- conn.ReadJSON(&msg):
				if err != nil {
					resHandler.errors <- errors.Wrap(err, "receive ws message")
					if errors.Is(err, websocket.ErrCloseSent) {
						return
					}
				}
			}
```

Reply:
Uh... I think it should. And looking at my original code... I can only say: 🤦

Reply:
Oh, wait. no, ReadJSON returns the error, it's not a chan... reverting.